dbms/src/DataStreams/CuttingUnionAllRequiredColumnBlockInputStream.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <Core/Block.h>
+#include <DataStreams/IProfilingBlockInputStream.h>
+
+namespace DB
+{
+
+/** Cutting block by required columns.
+  * Use required columns to cutting source blocks if required columns is not empty
+  *
+  * Example:
+  *     source_columns   : [X,X,A,B,X]
+  *     required_columns : [X,A,X]
+  *     output_columns   : [X,A,X]
+  */
+class CuttingUnionAllRequiredColumnBlockInputStream : public IProfilingBlockInputStream
```

Comment:
Because Interpreter cannot remove second `x` of second sub_union_query  for 
```
SELECT field_1,field_3,field_4 FROM (SELECT field_1,field_2,field_3,field_4 FROM table_a UNION ALL SELECT X,X,A,X FROM b);
```